<!DOCTYPE html>
<html lang="en">

<head>
  <meta charset="utf-8">
  <meta content="width=device-width, initial-scale=1.0" name="viewport">

  <title>ViralFlow</title>
  <meta content="" name="description">
  <meta content="" name="keywords">

  <!-- Favicons -->
  <link href="assets/img/favicon.png" rel="icon">
  <link href="assets/img/apple-touch-icon.png" rel="apple-touch-icon">

  <!-- Google Fonts -->
  <link href="https://fonts.googleapis.com/css?family=Open+Sans:300,300i,400,400i,600,600i,700,700i|Jost:300,300i,400,400i,500,500i,600,600i,700,700i|Poppins:300,300i,400,400i,500,500i,600,600i,700,700i" rel="stylesheet">

  <!-- Vendor CSS Files -->
  <link href="assets/vendor/aos/aos.css" rel="stylesheet">
  <link href="assets/vendor/bootstrap/css/bootstrap.min.css" rel="stylesheet">
  <link href="assets/vendor/bootstrap-icons/bootstrap-icons.css" rel="stylesheet">
  <link href="assets/vendor/boxicons/css/boxicons.min.css" rel="stylesheet">
  <link href="assets/vendor/glightbox/css/glightbox.min.css" rel="stylesheet">
  <link href="assets/vendor/remixicon/remixicon.css" rel="stylesheet">
  <link href="assets/vendor/swiper/swiper-bundle.min.css" rel="stylesheet">

  <!-- Template Main CSS File -->
  <link href="assets/css/style.css" rel="stylesheet">

  <!-- =======================================================
  * Template Name: Arsha - v4.9.0
  * Template URL: https://bootstrapmade.com/arsha-free-bootstrap-html-template-corporate/
  * Author: BootstrapMade.com
  * License: https://bootstrapmade.com/license/
  ======================================================== -->
</head>

<body>

  <!-- ======= Header ======= -->
  <header id="header" class="fixed-top ">
    <div class="container d-flex align-items-center">

      <h1 class="logo me-auto"><a href="index.html">ViralFlow</a></h1>
      <!-- Uncomment below if you prefer to use an image logo -->
      <!-- <a href="index.html" class="logo me-auto"><img src="assets/img/logo.png" alt="" class="img-fluid"></a>-->

      <nav id="navbar" class="navbar">
        <ul>
          <li><a class="nav-link scrollto active" href="#hero">Home</a></li>
          <li><a class="nav-link scrollto" href="#sobre">About</a></li>
          <li><a class="nav-link scrollto" href="#dependencias">Dependencies</a></li>
          <li><a class="nav-link scrollto" href="#instalacao">Install</a></li>
          <li><a class="nav-link scrollto" href="#guia">Quick Guide</a></li>
          <li><a class="nav-link scrollto" href="#argumentos">Parameters</a></li>
          <li><a class="nav-link scrollto" href="#outputs">Outputs</a></li>
          <li><a class="getstarted scrollto" href="#faq">FAQ</a></li>
          <li><a class="nav-link scrollto" href="index.html"><img src="assets/img/english-language.png" class="img-flag"></a></li>
        </ul>
        <i class="bi bi-list mobile-nav-toggle"></i>
      </nav><!-- .navbar -->

    </div>
  </header><!-- End Header -->

  <!-- ======= Hero Section ======= -->
  <section id="hero" class="d-flex align-items-center">

    <div class="container">
      <div class="row">
        <div class="col-lg-6 d-flex flex-column justify-content-center pt-4 pt-lg-0 order-2 order-lg-1" data-aos="fade-up" data-aos-delay="200">
          <h1>A Fiocruz Genomic Network Technology</h1>
          <h2>Built to viral analysis guided by reference (version v1.1.0).</h2>
          <div class="d-flex justify-content-center justify-content-lg-start">
            <a href="#guia" class="btn-get-started scrollto">Quick Guide</a>
          </div>
        </div>
        <div class="col-lg-6 order-1 order-lg-2 hero-img" data-aos="zoom-in" data-aos-delay="200">
          <img src="assets/img/viralflow_logo.png" class="img-home" alt="" style="object-fit:contain;width: 400px;height: 400px;">
        </div>
      </div>
    </div>

  </section><!-- End Hero -->

  <main id="main">

    <!-- ======= Sobre Section ======= -->
    <section id="sobre" class="about">
      <div class="container" data-aos="fade-up">

        <div class="section-title">
          <h2>About</h2>
        </div>
            <p>
            ViralFlow is a workflow developed for viral genomic surveillance that performs several genomic analyses based on reference genome assembly.<br>
            <br>
            The current version of the pipeline was written in the Nextflow workflow language, and can be applied to different viruses. Only a single 1 command line is needed to run the entire workflow. To make using ViralFlow more accessible, an installing “wrapper” was written to make the pipeline easier for people unfamiliar with Nextflow.<br>
            <br>
            Currently, the code has been tested only for Illumina paired-end data. The ViralFlow development team can not provide support for errors/problems of applying ViralFlow on sequencing reads from other platforms.<br>
            <br>
            The publication of the first version of ViralFlow can be checked <a href="https://www.mdpi.com/1999-4915/14/2/217" target="_blank">at this link</a>. And the update publication, which encompasses the functions described in this documentation, is in development.
            </p>
            <img src="assets/img/flowchart_en.png" class="img-home align-items-center" alt="">

            <div class="section-title">
              <h2>Intrahost Analysis</h2>
            </div>
                <p>
                  ViralFlow presents its own algorithm to detect iSNV (intrahost Single Nucleotide Variant) regions.<br>
                  <br>
                  For a given multiallelic site to have a lower frequency allele considered as iSNV, three conditions must be met, these conditions are schematized in the upper part of the figure below.
                  <br>
                  <br>
                  n this logic, the bottom part of the figure below represents 3 multi-allelic sites and exemplifies the logic for considering which of them should be considered as iSNVs.<br>
                  <br>
                </p>
                <img src="assets/img/viralflow_snv.png" class="img-home align-items-center" alt="">
              </div>
            </div>
    
          </div>
      </div>
    </section><!-- End Sobre Section -->

    <!-- ======= Dependencias Section ======= -->
    <section id="dependencias" class="why-us section-bg">
      <div class="container" data-aos="fade-up">
        <div class="section-title">
          <h2>dependencies</h2>
        </div>
            <p>
            To install and run ViralFlow, you will need at least 4.4Gb of disk storage (to build containers), and three dependencies:
            </p>
            <ul>
              <li>pip</li>
              <li>Git</li>
              <li>uidmap</li>
            </ul>
            <p>
              The list with tool versions used in ViralFlow can be checked <a href="https://github.com/WallauBioinfo/ViralFlow/tree/main/versions" target="_blank">at this link</a>. Moreover, for an installation with a controlled environment, we use Conda. ViralFlow uses the Singularity containerization tool and the Nextflow workflow language. To ensure that ViralFlow works correctly, the correct versions of Singularity and Nextflow are essential, both of which can be installed together with ViralFlow.
            </p>
            <br>
            <div class="accordion-list">
              <ul>
                <li>
                  <a data-bs-toggle="collapse" data-bs-target="#accordion-list-1" class="collapsed"><span>01</span> Conda
                    <i class="bx bx-chevron-down icon-show"></i><i class="bx bx-chevron-up icon-close"></i></a>
                  <div id="accordion-list-1" class="collapse" data-bs-parent=".accordion-list">
                    <p>
                      Conda is an open source package manager. It was initially developed to manage Python language package environments, allowing version control.<br>
                      <br>
                      So that all users can use ViralFlow with the exact versions of Singularity and Nextflow, we created an environment with these versions, which allows a standardization in the ViralFlow installation environments, as well as the resolution of eventual errors.<br>
                    </p>
                  </div>
                </li>
                <li>
                  <a data-bs-toggle="collapse" data-bs-target="#accordion-list-2" class="collapsed"><span>02</span> Singularity v3.11.4
                    <i class="bx bx-chevron-down icon-show"></i><i class="bx bx-chevron-up icon-close"></i></a>
                  <div id="accordion-list-2" class="collapse" data-bs-parent=".accordion-list">
                    <p>
                      Singularity is a tool that performs virtualization at the operating system level, a task popularly known as containerization.<br>
                      <br>
                      For a computational pipeline like ViralFlow to work, it requires several dependencies (other tools and libraries of the operating system) that can be complex to install. In order to make things easier to the user we use the container system which already comes with all dependencies needed to run ViralFlow. Therefore, the user does not need to install everything on his own.<br>
                      <br>
                      With containerization, the user only needs to install one tool, in this case Singularity, and then build the containers based on recipe files (in the case of ViralFlow, this was also automated with the setupContainers.sh script). In this way, different research groups can run the tool with the same computational environment, without changes in ViralFlow's behavior, which guarantees the reproducibility of the results.
                    </p>
                  </div>
                </li>
                <li>
                  <a data-bs-toggle="collapse" data-bs-target="#accordion-list-3" class="collapsed"><span>03</span> Nextflow 22.04 <i class="bx bx-chevron-down icon-show"></i><i class="bx bx-chevron-up icon-close"></i></a>
                  <div id="accordion-list-3" class="collapse" data-bs-parent=".accordion-list">
                    <p>
                      Nextflow is a bioinformatics workflow manager that allows the development of portable and reproducible workflows.<br>
                      <br>
                      Nextflow allows the execution of workflows in different computing environments, such as on-premises, high-performance computing (HPC) environments, cloud computing services, such as AWS and Google Cloud, and also in container orchestration systems, such as Kubernetes.<br>
                      <br>
                      In addition, Nextflow provides support for managing dependencies with Conda, Docker, Podman or Singularity.                      
                    </p>
                  </div>
                </li>
              </ul>
            </div>
        </div>
      </div>
    </section><!-- End Dependencias Section -->

    <!-- ======= Instalação Section ======= -->
    <section id="instalacao" class="skills">
      <div class="container" data-aos="fade-up">

        <div class="section-title">
          <h2>How to install (MacOS)?</h2>
        </div>
          <p>
            Due to the limitation of using singularity on MacOS, to run ViralFlow on this type of system, we suggest using a Linux virtualization software called <a href="https://github.com/lima-vm/lima">Lima</a>. In this way, the user must follow three steps for installing Lima, and then satisfied with the Ubuntu installation guide.</p>
          <h3>Installing Lima</h3>
          <p>
            Install Lima using the Homebrew package manager:
          </p>
          <div class="container-code">
            <div class="row-code">
              <div class="column left">
                <span class="dot" style="background:#ED594A;"></span>
                <span class="dot" style="background:#FDD800;"></span>
                <span class="dot" style="background:#5AC05A;"></span>
              </div>
            </div>
            <div class ="box-code">
              <p color = "green">
                brew install lima
              </p>
            </div>
            <br>
          </div>
          <h3>Installing the Ubuntu instance</h3>
          <p>
            Follow the step by step installation of an Ubuntu instance by entering the command below
          </p>
          <div class="container-code">
            <div class="row-code">
              <div class="column left">
                <span class="dot" style="background:#ED594A;"></span>
                <span class="dot" style="background:#FDD800;"></span>
                <span class="dot" style="background:#5AC05A;"></span>
              </div>
            </div>
            <div class ="box-code">
              <p color = "green">
                limactl start
              </p>
            </div>
          </div>
          <br>
          <h3>Start Ubuntu</h3>
          <p>
            When the virtual machine is installed, use the command below to start Ubuntu and follow the ViralFlow installation steps as usual.
          </p>
          <div class="container-code">
            <div class="row-code">
              <div class="column left">
                <span class="dot" style="background:#ED594A;"></span>
                <span class="dot" style="background:#FDD800;"></span>
                <span class="dot" style="background:#5AC05A;"></span>
              </div>
            </div>
            <div class ="box-code">
              <p color = "green">
                lima
              </p>
            </div>
          </div>
          <br>
        </div>
      <div class="container" data-aos="fade-up">

        <div class="section-title">
          <h2>How to install (Ubuntu)?</h2>
        </div>
          <p>
            To install ViralFlow, four steps are necessary: Install system dependencies, in case you haven't installed them; Install Conda; install ViralFlow and, finally, assemble the containers for the analyses. This process is performed only once.
            <br>
            <br>
            ViralFlow was developed and tested for the following operational systems and versions:
            <ul>
              <li> Ubuntu 20.04 LTS;</li>
              <li> Ubuntu 22.04 LTS.</li>
            </ul>
          </p>
          <h3>Installing system dependencies</h3>
          <p>
            If you don't have the pip dependency installer, the git version control system, and the uidmap package, you must install it with the following lines:
          </p>
          <div class="container-code">
            <div class="row-code">
              <div class="column left">
                <span class="dot" style="background:#ED594A;"></span>
                <span class="dot" style="background:#FDD800;"></span>
                <span class="dot" style="background:#5AC05A;"></span>
              </div>
            </div>
            <div class ="box-code">
              <p color = "green">
                sudo apt update -y && \<br>
                &emsp;&emsp;sudo apt upgrade -y && \<br>
                &emsp;&emsp;sudo apt install curl git python3-pip uidmap -y 
              </p>
            </div>
            <br>
          </div>
          <h3>Installing and configuring the micromamba environment</h3>
          <p>
            We recommend managing micromamba environments due to their parallelization when downloading and installing dependencies. If you use another environment manager, such as conda, miniconda, mamba, etc., you can continue installing ViralFlow by skipping this step, however, conflicts during installation may occur due to versions of the specified dependencies, and their respective availability on different channels of resources.<br>
            <br>
            If you don't have the micromamba environment manager installed, you can install and configure it with the following lines:
          </p>
          <div class="container-code">
            <div class="row-code">
              <div class="column left">
                <span class="dot" style="background:#ED594A;"></span>
                <span class="dot" style="background:#FDD800;"></span>
                <span class="dot" style="background:#5AC05A;"></span>
              </div>
            </div>
            <div class ="box-code">
              <p color = "green">
                cd $HOME<br>
                curl -Ls https://micro.mamba.pm/api/micromamba/linux-64/1.5.7 | tar -xvj bin/micromamba<br>
                ./bin/micromamba shell init -s bash -p ~/micromamba<br>
                source ~/.bashrc<br>
                micromamba activate<br>
              </p>
            </div>
          </div>
          <br>
          <h3>Installing ViralFlow</h3>
          <p>
            If you already have the aforementioned dependencies and conda installed, you can install ViralFlow with 5 lines of code:
          </p>
          <div class="container-code">
            <div class="row-code">
              <div class="column left">
                <span class="dot" style="background:#ED594A;"></span>
                <span class="dot" style="background:#FDD800;"></span>
                <span class="dot" style="background:#5AC05A;"></span>
              </div>
            </div>
            <div class ="box-code">
              <p color = "green">
                git clone https://github.com/WallauBioinfo/ViralFlow <br>
                cd ViralFlow/ <br>
                micromamba env create -f envs/env.yml <br>
                micromamba activate viralflow <br>
                pip install -e . <br>
              </p>
            </div>
          </div>
          <br>
          <h3>Building the containers.</h3>
          <p>
            All steps of ViralFlow are performed in controlled environments, where each tool will have the same version regardless of the research group that is using the tool. ViralFlow has its own method to carry out all this construction of environments, running just one line of code.
            For the building of containers,  ViralFlow requires that the tool "unsquashfs" be available in the directory "/usr/local/bin/". During the container building process, a step to check for the presence of unsquashfs is performed.
            If the user's system does not have "unsquashfs" in the ideal location, ViralFlow will suggest the steps necessary to solve this problem.
            To avoid this, the user can create a symbolic link of the "unsquashfs" tool to the directory "/usr/local/bin/", for this the following command must be executed:
          </p>
          <div class="container-code">
            <div class="row-code">
              <div class="column left">
                <span class="dot" style="background:#ED594A;"></span>
                <span class="dot" style="background:#FDD800;"></span>
                <span class="dot" style="background:#5AC05A;"></span>
              </div>
            </div>
            <div class ="box-code">
              <p color = "green">
                sudo ln -s /usr/bin/unsquashfs /usr/local/bin/unsquashfs<br>
              </p>
            </div>
            <p>
              After ensuring that "unsquashfs" is in the appropriate location, run the command to build the containers:</p>
            <div class="containers-code">
              <div class="row-code">
                <div class="column left">
                  <span class="dot" style="background:#ED594A;"></span>
                  <span class="dot" style="background:#FDD800;"></span>
                  <span class="dot" style="background:#5AC05A;"></span>
                </div>
              </div>
              <div class ="box-code">
                <p color = "green">
                  viralflow -build_containers<br>
                </p>
              </div>
          </div>
        </div>
        <br>
    </section><!-- End Instalação Section -->

    <!-- ======= Guia Section ======= -->
    <section id="guia" class="skills">
      <div class="container" data-aos="fade-up">
        <div class="section-title">
          <h2>How to run (Quick Guide)</h2>
          <p align="justify">ViralFlow provides 2 usage modes: sars-cov2 and custom. Regardless of the mode, the user must provide the absolute paths (the entire path to the directory or file to be indicated for the pipeline eg /home/user/test/) for each input file in the command line, otherwise the pipeline will be interrupted during execution. The explanation of each argument <a href="https://viralflow.github.io/index-en.html#argumentos">can be found here</a>.</p>
          <br>
          <div class="row">
            <div class="content" data-aos="fade-left" data-aos-delay="100">
            <h3 align="justify"> Customizing the snpEff database </h3>
            </div>
            <div class="col-lg-6 pt-4 pt-lg-0 content" data-aos="fade-left" data-aos-delay="100">
                <p align="justify">
                By default, the snpEff tool in ViralFlow is configured with the NC_045512.2 genome of the SARS-CoV-2 virus only. If you want to include the snpEff analysis for other viruses, you must update the snpEff database with the following line, example with Dengue:</p><br>
            </div>
            <div class="col-lg-6 d-flex" data-aos="fade-right" data-aos-delay="100">
              <div class="container-code" style="margin-top:10px !important">
                <div class="row-code">
                  <div class="column left" align="right">
                    <span class="dot" style="background:#ED594A;"></span>
                    <span class="dot" style="background:#FDD800;"></span>
                    <span class="dot" style="background:#5AC05A;"></span>
                  </div>
                </div>
                <div class ="box-code">
                  <p color = "green" align="left">
                    viralflow -add_entry_to_snpeff --org_name Dengue --genome_code NC_001474.2             
                  </p>
                </div>
              </div>
            </div>
          </div>
          <br>
          <div class="row">
            <div class="content" data-aos="fade-left" data-aos-delay="100">
              <h3 align="justify"> sars-cov2  </h3>
            </div>
            <div class="col-lg-6 pt-4 pt-lg-0 content" data-aos="fade-left" data-aos-delay="100">
                <p align="justify">
                  In this model, the analysis is performed based on the reference genome NC_045512.2, and has, as additional analysis, the signature of strains with the Pangolin tool, and signing clades and mutations with the Nextclade tool. For this, the user needs to build a file with the analysis parameters, an example can be <a href="https://github.com/WallauBioinfo/ViralFlow/blob/main/test_files/sars-cov-2.params">seen here</a>.
                </p><br>          
            </div>
            <div class="col-lg-6 d-flex" data-aos="fade-right" data-aos-delay="100">
              <div class="container-code">
                <div class="row-code">
                  <div class="column left" align="right">
                    <span class="dot" style="background:#ED594A;"></span>
                    <span class="dot" style="background:#FDD800;"></span>
                    <span class="dot" style="background:#5AC05A;"></span>
                  </div>
                </div>
                <div class ="box-code">
                  <p color = "green" align="left">
                    viralflow -run --params_file test_files/sars-cov-2.params  
                  </p>
                </div>
              </div>
            </div>
          </div>
          <br>
          <div class="row">
            <div class="content" data-aos="fade-left" data-aos-delay="100">
              <h3 align="justify"> custom </h3>
            </div>
            <div class="col-lg-6 pt-4 pt-lg-0 content" data-aos="fade-left" data-aos-delay="100">
                <p align="justify">
                In this model, the analysis is performed based on the files for the virus that the user wants to analyze. In this mode, the user is responsible for providing each of the files necessary for the analysis. If the user wants to perform the snpEff analysis, he must pass the refseq code viral genome</p><br>     
            </div>
            <div class="col-lg-6 d-flex" data-aos="fade-right" data-aos-delay="100">
              <div class="container-code">
                <div class="row-code">
                  <div class="column left" align="right">
                    <span class="dot" style="background:#ED594A;"></span>
                    <span class="dot" style="background:#FDD800;"></span>
                    <span class="dot" style="background:#5AC05A;"></span>
                  </div>
                </div>
                <div class ="box-code">
                  <p color = "green" align="left">
                    viralflow -run --params_file test_files/denv.params         
                  </p>
                </div>
              </div>
            </div>
          </div>
          <br>
          <div class="row">
            <div class="content" data-aos="fade-left" data-aos-delay="100">
              <h3 align="justify"> Pangolin update </h3>
            </div>
            <div class="col-lg-6 pt-4 pt-lg-0 content" data-aos="fade-left" data-aos-delay="100">
                <p align="justify">
                  Periodically the pangolin tool updates the lineage database, as well as the usher classification phylogeny, the scorpion mutation constellations, and the pangoLearn trained model. To update the tool and/or it's databases, just run ViralFlow with one of the commands:</p><br>     
            </div>
            <div class="col-lg-6 d-flex" data-aos="fade-right" data-aos-delay="100">
              <div class="container-code">
                <div class="row-code">
                  <div class="column left" align="right">
                    <span class="dot" style="background:#ED594A;"></span>
                    <span class="dot" style="background:#FDD800;"></span>
                    <span class="dot" style="background:#5AC05A;"></span>
                  </div>
                </div>
                <div class ="box-code">
                  <p color = "green" align="left">
                    #update the tool and databases<br>
                    viralflow -update_pangolin<br>
                    <br>
                    #update only the tool<br>
                    viralflow -update_pangolin_data<br>                             
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </section><!-- End Guia Section -->

    <!-- ======= Argumentos Section ======= -->
    <section id="argumentos" class="cta">
      <div class="container" data-aos="zoom-in">
        <div class="section-title">
          <h2>Arguments</h2>
        <table id="tables">
          <tr>
            <th>Argument</th>
            <th>Default value</th>
            <th>Description</th>
          </tr>
          <tr>
            <td>virus</td>
            <td>sars-cov2</td>
            <td>Analysis type (sars-cov2 or custom).</td>
          </tr>
          <tr>
            <td>primersBED</td>
            <td>null</td>
            <td>Absolute path to bed file with primers information used in genomic amplification (optional).</td>
          </tr>
          <tr>
            <td>outDir</td>
            <td>launchDir/output/</td>
            <td>Absolute path to the directory where the results will be stored.</td>
          </tr>
          <tr>
            <td>inDir</td>
            <td>launchDir/input/</td>
            <td>Absolute path to the directory with the input data (directory with the FASTQ files).</td>
          </tr>
          <tr>
            <td>runSnpEff</td>
            <td>true</td>
            <td>Needed to run the snpEff tool (true or false).</td>
          </tr>
          <tr>
            <td>writeMappedReads</td>
            <td>true</td>
            <td>Needed to generate the FASTQ files containing the sequencing reads that mapped to the reference genome.</td>
          </tr>
          <tr>
            <td>minLen</td>
            <td>75</td>
            <td>Minimum size the reads must have. Reads below this thershould will be eliminated by FastP.</td>
          </tr>
          <tr>
            <td>depth</td>
            <td>5</td>
            <td>Minimum coverage depth to call consensus bases. Positions with lower coverage depth will not be called and a “-” will be added to the respective consensus genomic position.</td>
          </tr>
          <tr>
            <td>mapping_quality</td>
            <td>30</td>
            <td>Mapping quality treshold used to variant calling.</td>
          </tr>
          <tr>
            <td>base_quality</td>
            <td>30</td>
            <td>Base quality treshold used to variant calling.</td>
          </tr>
          <tr>
            <td>minDpIntrahost</td>
            <td>100</td>
            <td>Minimum coverage depth per genomic site to be considered in the intrahost analysis.</td>
          </tr>
          <tr>
            <td>trimLen</td>
            <td>0</td>
            <td>Number of bases that should be trimmed at both ends of the reads.</td>
          </tr>
          <tr>
            <td>refGenomeCode</td>
            <td>null</td>
            <td>Code of the genome to be used in the custom analysis.</td>
          </tr>
          <tr>
            <td>referenceGFF</td>
            <td>null</td>
            <td>GFF genome file to be used in custom analysis.</td>
          </tr>
          <tr>
            <td>referenceGenome</td>
            <td>null</td>
            <td>Fasta genome file to be used in custom analysis.</td>
          </tr>
          <tr>
            <td>nextflowSimCalls</td>
            <td>null</td>
            <td>Number of simultaneous calls that nextflow can perform.</td>
          </tr>
          <tr>
            <td>fastp_threads</td>
            <td>1</td>
            <td>Number of threads to be used in the fastp read filtering step.</td>
          </tr>
          <tr>
            <td>bwa_threads</td>
            <td>1</td>
            <td>Number of threads to be used in the bwa mapping step.</td>
          </tr>
          <tr>
            <td>mafft_threads</td>
            <td>1</td>
            <td>Number of threads to be used in the mafft alignment step.
          </tr>
        </table>
      </div>
    </div>
    </section><!-- End Argumentos Section -->

    <!-- ======= Outputs Section ======= -->
    <section id="outputs" class="portfolio">
      <div class="container" data-aos="fade-up">
        <div class="section-title">
          <h2>Output files </h2>
          <p>The ViralFlow tool generates 2 types of output: specific and compiled.</p><br>
          <h3 align="left"> Specific </h3>
          <p align="justify">For each sample in the analysis, a results directory will be created with the pattern `prefix_results` where the prefix is ​​a code created from the fastq file name of the sample. Each directory has the following results:</p><br>
          <table id="tables">
            <tr>
              <th>File</th>
              <th>Description</th>
            </tr>
            <tr>
              <td>prefix.all.fa.pango.out.csv</td>
              <td>Tabular file with the results of the pangolin tool.</td>
            </tr>
            <tr>
              <td>prefix.ann.vcf</td>
              <td>file in tsv format (Tab separated value - comma-separated values) with annotation of the variants made by the snpEff tool.</td>
            </tr>
            <tr>
              <td>prefix.depth5.all.fa.nextclade.csv</td>
              <td>Tabular file with the results of the nextclade tool.</td>
            </tr>
            <tr>
              <td>prefix.depth5.amb.fa</td>
              <td>Consensus genome with ambiguous nucleotides at multiple allele positions.</td>
            </tr>
            <tr>
              <td>prefix.depth5.fa</td>
              <td>Consensus genome with majority nucleotides. Majority consensus, normally deposited in GISAID.</td>
            </tr>
            <tr>
              <td>prefix.depth5.fa.algn</td>
              <td>Consensus genome aligned with reference genome (following same size, mafft --keeplenght).</td>
            </tr>
            <tr>
              <td>prefix.depth5.fa.algn.minor.fa</td>
              <td>Consensus genome with minority nucleotides.</td>
            </tr>
            <tr>
              <td>prefixo.depth5.fa.bc.intrahost.short.tsv</td>
              <td>Tabular file summarizing the genomic positions where intrahost variants are supported.</td>
            </tr>
            <tr>
              <td>prefix.depth5.fa.bc.intrahost.tsv</td>
              <td>Tabular file with all information on intrahost variant positions.</td>
            </tr>
            <tr>
              <td>prefix.mapped.R1.fq.gz</td>
              <td>FASTQ R1 file with mapped reads.</td>
            </tr>
            <tr>
              <td>prefix.mapped.R2.fq.gz</td>
              <td>FASTQ R2 file with mapped reads.</td>
            </tr>
            <tr>
              <td>prefix.metrics.genome.tsv</td>
              <td>Tabular file with mapping depth and coverage metrics.</td>
            </tr>
            <tr>
              <td>prefix.fastp.html</td>
              <td>HTML file summarizing the results of the fastp tool.</td>
            </tr>
            <tr>
              <td>prefix_snpEff_summary.html</td>
              <td>HTML file summarizing the results of the snpEff tool.</td>
            </tr>
            <tr>
              <td>prefix.sorted.bam</td>
              <td>File with the sorted mapping of the sample reads against the reference genome.</td>
            </tr>
            <tr>
              <td>prefix.tsv</td>
              <td>vcf-like file generated by iVar.</td>
            </tr>
            <tr>
              <td>prefix.unmapped.R1.bam.fq</td>
              <td>FASTQ R1 file with unmapped reads.</td>
            </tr>
            <tr>
              <td>prefix.unmapped.R2.bam.fq</td>
              <td>FASTQ R2 file with unmapped reads.</td>
            </tr>
            <tr>
              <td>prefix.vcf</td>
              <td>vcf file.</td>
            </tr>
            <tr>
              <td>metrics.alignment_summary_metrics</td>
              <td>Text file with a summary of various metrics from the mapping.</td>
            </tr>
            <tr>
              <td>nextclade.errors.csv</td>
              <td>File reporting nextclade errors.</td>
            </tr>
            <tr>
              <td>nextclade_gene_*.translation.fasta</td>
              <td>File with the proteins of each gene.</td>
            </tr>
            <tr>
              <td>snpEff_genes.txt</td>
              <td>Tabular file with the number of variants and estimated impact per gene.</td>
            </tr>
            <tr>
              <td>wgs</td>
              <td>Textual file with mapping metrics, including depth by region.</td>
            </tr>
            <tr>
              <td>prefixo_coveragePlot.png</td>
              <td>PNG file with graphical visualization of genome coverage.</td>
            </tr>
            <tr>
              <td>prefix_coveragePlot.svg</td>
              <td>SVG file with graphical visualization of genome coverage.</td>
            </tr>
            <tr>
              <td>prefix_coveragePlot.html</td>
              <td>HTML file  com a visualização gráfica da cobertura do genoma.</td>
            </tr>
            <tr>
              <td>prefix_snpPlot.png</td>
              <td>PNG file with a graphical visualization of the SNPs detected in the sample relative to the reference genome.</td>
            </tr>
            <tr>
              <td>prefix_snpPlot.svg</td>
              <td>SVG file with a graphical visualization of the SNPs detected in the sample relative to the reference genome.</td>
            </tr>
          </table>
          <br>
          <br>
          <h3 align="left">Compiled</h3>
          <p align="justify">At the end of the analysis, a directory named COMPILED_OUTPUT is generated with the compiled results, where:</p><br>
          <table id="tables">
            <tr>
              <th>File</th>
              <th>Description</th>
            </tr>
            <tr>
              <td>errors_detected.csv</td>
              <td>Samples with no resulting consensus.</td>
            </tr>
            <tr>
              <td>lineage_summary.csv</td>
              <td>It is generated for sars-cov- 2, presenting the lineage counts identified in the analysis, including lineages from minority genomes.</td>
            </tr>
            <tr>
              <td>major_summary.csv</td>
              <td>Tabular file with a summary of the depth, coverage and lineages (in case of sars-cov-2 analyses) of the majority consensus.</td>
            </tr>
            <tr>
              <td>minor_summary.csv</td>
              <td>Tabular file with a summary of the depth, coverage and lineages (in case of sars-cov-2 analyses) of minority consensus.</td>
            </tr>
            <tr>
              <td>mutations.csv</td>
              <td>Tabular file with the mutations found.</td>
            </tr>
            <tr>
              <td>nextclade.csv</td>
              <td>Tabular file with the result of nextclade (in case of sars-cov-2 scans).</td>
            </tr>
            <tr>
              <td>pango.csv</td>
              <td>Tabular file with the result of pangolin (in case of sars-cov-2 scans).</td>
            </tr>
            <tr>
              <td>reads_count.csv</td>
              <td>Tabular file with the count of reads per sample.</td>
            </tr>
            <tr>
              <td>seqbatch.fa</td>
              <td>Fasta file with majority consensus.</td>
            </tr>
            <tr>
              <td>short_summary.csv</td>
              <td>Tabular file with a summary of depth, coverage and lineage (in case of sars-cov-2 analyses) of both consensus.<br>
              <br>
              <ul>
                <li>cod: Prefix defined for each sample;</li>
                <li>mean_depth_coverag: Mean coverage depth;</li>
                <li>sd_depth_coverage: Standart deviation of mean coverage depth;</li>
                <li>median_depth_coverage: Median coverage depth;</li>
                <li>taxon: Prefix sample with consensus code (major or minor);</li>
                <li>lineage (only sars-cov-2): Viral lineage;</li>
                <li>scorpio_call (only sars-cov-2): Virus variant;</li>
                <li>coverage_breadth: COverage breath of assembly.</li>
              </ul></td>
            </tr>
            <tr>
              <td>wgs.csv</td>
              <td>Tabular file with picard output.</td>
            </tr>
            <tr>
              <td>VF_REPORT/index.html</td>
              <td>Summary HTML file with the compilation of fastp and snpEff results.</td>
            </tr>
          </table>
        </div>
        </div>
      </div>
    </section><!-- End Output Section -->

    <!-- ======= Frequently Asked Questions Section ======= -->
    <section id="faq" class="portfolio">
      <div class="container" data-aos="fade-up">
        <div class="section-title">
          <h2>Frequently Asked Questions</h2></div>
          <h4>Should I mount containers for each analysis?</h4>
            <p>No, containers should only be rebuilt if there is an update of the workflow or specific tool of the pipeline (i.e. pangolin).</p>
          <h4>ViralFlow returned an error due to the absence of "unsquashfs" in the directory "/usr/local/bin/". What should I do?</h4>
            <p>A symbolic link for "unsquashfs" must be created. To do this, run the following command: sudo ln -s /usr/bin/unsquashfs /usr/local/bin/unsquashfs</p>
          <h4>Should I provide the adapters fasta file?</h4>
            <p>You can provide the fasta file with the PCR primers and/or adaptors in analyzes that use PCR amplification of viral genetic material prior to sequencing.</p>
          <h4>My ViralFlow review reported an error in the Pangolin update step, why? </h4>
            <p>This may happen if the Pangolin team releases a version with new dependencies, in which case you must rebuild the pangolin container with the line: singularity pull docker://staphb/pangolin</p>
          <h4>My installation went wrong in the step of mounting the containers, why?</h4>
            <p>What might have happened is that you have a different version of singularity than v3.11.4, you can check that with singularity --version. Another factor that can cause errors is that your internet network blocks query domains for building containers.</p>
          <h4>I installed the tool, my files are all organized following the manual, but the samples are not found, why?</h4>
            <p>ViralFlow recognizes samples by the suffixes _R1 or _R2 followed by .fq.gz or .fastq.gz, make sure your samples follow the required pattern. Examples:
              <ul>
                <li>ART1_R1.fq.gz, ART1_R2.fq.gz will be recognized as the ART1 sample;</li>
                <li>ART1_denv_R1.fastq.gz, ART1_denv_R2.fastq.gz will be recognized as the ART1_denv sample.</li>
              </ul>
            </p>
            <h4>Can I run ViralFlow on Windows??</h4>
             <p>Although some versions of Windows systems integrate a Linux interface through WSL (Windows Subsystem for Linux), our development team has not tested ViralFlow in these environments, therefore, the user is free to test the workflow in this type of environment, However, our team will not provide support for errors arising from these tests.</p>
      </div>
    </section><!-- End Frequently Asked Questions Section -->

  <!-- ======= Footer ======= -->
  <footer id="footer">

    <div class="container footer-bottom clearfix">
      <div class="container">
        <h3>Wallau Lab</h3>
        <p>
          Fundação Oswaldo Cruz, Instituto Aggeu Magalhães<br>
          Universidade Federal de Pernambuco - Campus da, Av. Prof. Moraes Rego, s/n<br>
          Recife, Brasil <br><br>
          <strong>Site:</strong> https://wallaulab.wordpress.com<br>
        </p>
      </div>
      <div class="copyright">
        &copy;<strong><span>WallauLab</span></strong>. All rights reserved.
      </div>
      <div class="credits">
        <!-- All the links in the footer should remain intact. -->
        <!-- You can delete the links only if you purchased the pro version. -->
        <!-- Licensing information: https://bootstrapmade.com/license/ -->
        <!-- Purchase the pro version with working PHP/AJAX contact form: https://bootstrapmade.com/arsha-free-bootstrap-html-template-corporate/ -->
        Designed by <a href="https://bootstrapmade.com/">BootstrapMade</a><br>
        Edited by <a href="https://dezordi.github.io/">Filipe Dezordi</a>
      </div>
    </div>
  </footer><!-- End Footer -->

  <div id="preloader"></div>
  <a href="#" class="back-to-top d-flex align-items-center justify-content-center"><i class="bi bi-arrow-up-short"></i></a>

  <!-- Vendor JS Files -->
  <script src="assets/vendor/aos/aos.js"></script>
  <script src="assets/vendor/bootstrap/js/bootstrap.bundle.min.js"></script>
  <script src="assets/vendor/glightbox/js/glightbox.min.js"></script>
  <script src="assets/vendor/isotope-layout/isotope.pkgd.min.js"></script>
  <script src="assets/vendor/swiper/swiper-bundle.min.js"></script>
  <script src="assets/vendor/waypoints/noframework.waypoints.js"></script>

  <!-- Template Main JS File -->
  <script src="assets/js/main.js"></script>

</body>
</html>
